zenpy/lib/util.py

```diff
@@ -114,6 +114,13 @@ def id_of(x):
 
     return outer
 
+def dict_clean(items):
+    result = {}
+    for key, value in items:
+        if value is None:
+            value = ''
+        result[key] = value
+    return result
```

Comment:
I like that this has been extracted in util and we don't chew this data in the method itself.

How about renaming it to `dict_replace_none_with(items, replacement='')`?

A bit more specific, more suitable for reusable util functions?

Reply:
Still Minor (your call, easy): How about?

```py
def dict_replace(items, old_value, new_value=''):
    result = {}
    for key, value in items:
        if value == old_value:
            value = new_value
        result[key] = value
    return result
```

A bit clearer than `clean`, more suitable for reusable util functions?

And some free unit tests from a well known AI:


```py
import unittest

class TestDictReplace(unittest.TestCase):

    def test_dict_replace(self):
        self.assertEqual(dict_replace([('key1', None), ('key2', 'value2')], None, 'default'), {'key1': 'default', 'key2': 'value2'})
        self.assertEqual(dict_replace([('key1', None), ('key2', None)], None, 'default'), {'key1': 'default', 'key2': 'default'})
        self.assertEqual(dict_replace([('key1', 'value1'), ('key2', 'value2')], None, 'default'), {'key1': 'value1', 'key2': 'value2'})
        self.assertEqual(dict_replace([('key1', None), ('key2', 'value2')], None), {'key1': '', 'key2': 'value2'})
        self.assertEqual(dict_replace([('key1', None), ('key2', None)], None), {'key1': '', 'key2': ''})
        self.assertEqual(dict_replace([('key1', 'value1'), ('key2', 'value2')], None), {'key1': 'value1', 'key2': 'value2'})

if __name__ == '__main__':
    unittest.main()
```

Reply:
https://github.com/facetoe/zenpy/pull/579/files#r1323802568 
Could build a layer of clean functions.
dict_clean
followed by the dict_clean_None?